### ISSUE TITLE
Failed to compile `undefined symbol: sqlite3_db_config` + `unresolved imports [...]`

### ISSUE THREAD

**Issue by fooryo** [ffplayout/ffplayout#580]
**Describe the bug**

Failed to compile ona Manjaro Linux via AUR package `ffplayout` 

```
`error[E0432]: unresolved imports `self::encode::Encode`, `self::decode::Decode`, `super::Decode`, `super::Encode`, `super::FromRow`, `super::Type`
```

**To Reproduce**

1. Install AUR package `ffplayout` with your favourite helper or manually
2. Wait for the error

**Expected behavior**
It should at least compile

**Desktop/Server/Software (please complete the following information):**

 - OS: Manjaro Linux 6.7.7-1-MANJARO KDE Plasma 5.27.11
- ffplayout version 0.20.5-0

**Logging:**

```
   [...]
   Compiling argon2 v0.5.3
   Compiling sanitize-filename v0.5.0
   Compiling rpassword v7.3.1
   Compiling sysinfo v0.30.5
   Compiling sqlx v0.7.3
error: /home/user/.cache/paru/clone/ffplayout/src/ffplayout-0.20.5/target/release/deps/libsqlx_macros-f11e3188463e4adb.so: undefined symbol: sqlite3_db_config
  --> /home/user/.cache/paru/clone/ffplayout/src/rust-home/registry/src/index.crates.io-6f17d22bba15001f/sqlx-0.7.3/src/lib.rs:53:1
   |
53 | pub extern crate sqlx_macros;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0432]: unresolved imports `self::encode::Encode`, `self::decode::Decode`, `super::Decode`, `super::Encode`, `super::FromRow`, `super::Type`
   --> /home/user/.cache/paru/clone/ffplayout/src/rust-home/registry/src/index.crates.io-6f17d22bba15001f/sqlx-0.7.3/src/lib.rs:119:9
    |
119 | pub use self::encode::Encode;
    |         ^^^^^^^^^^^^^^^^^^^^
...
130 | pub use self::decode::Decode;
    |         ^^^^^^^^^^^^^^^^^^^^
...
144 |     pub use super::Decode;
    |             ^^^^^^^^^^^^^
145 |     pub use super::Encode;
    |             ^^^^^^^^^^^^^
146 |     pub use super::Executor;
147 |     pub use super::FromRow;
    |             ^^^^^^^^^^^^^^
...
151 |     pub use super::Type;
    |             ^^^^^^^^^^^

For more information about this error, try `rustc --explain E0432`.
error: could not compile `sqlx` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```

**Comment by jb-alvarado** [comment]
Hello @fooryo, what rust version have you installed? Maybe it needs a newer version.

@JunioCalu, you use also arch? Had you the same problem?

I use always the latest stable rust version and compile directly from repo and had never a problem.

**Comment by fooryo** [comment]
updated via `rustup toolchain install stable` to 1.77.1 (7cf61ebde 2024-03-27)

still :
```
[...]
   Compiling local-ip-address v0.6.0
   Compiling faccess v0.2.4
   Compiling path-clean v1.0.1
error: /home/salotto/.cache/paru/clone/ffplayout/src/ffplayout-0.20.5/target/release/deps/libsqlx_macros-f11e3188463e4adb.so: undefined symbol: sqlite3_db_config
  --> /home/salotto/.cache/paru/clone/ffplayout/src/rust-home/registry/src/index.crates.io-6f17d22bba15001f/sqlx-0.7.3/src/lib.rs:53:1
   |
53 | pub extern crate sqlx_macros;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

   Compiling relative-path v1.9.2
   Compiling zstd v0.13.0
error[E0432]: unresolved imports `self::encode::Encode`, `self::decode::Decode`, `super::Decode`, `super::Encode`, `super::FromRow`, `super::Type`
   --> /home/salotto/.cache/paru/clone/ffplayout/src/rust-home/registry/src/index.crates.io-6f17d22bba15001f/sqlx-0.7.3/src/lib.rs:119:9
    |
119 | pub use self::encode::Encode;
    |         ^^^^^^^^^^^^^^^^^^^^
...
130 | pub use self::decode::Decode;
    |         ^^^^^^^^^^^^^^^^^^^^
...
144 |     pub use super::Decode;
    |             ^^^^^^^^^^^^^
145 |     pub use super::Encode;
    |             ^^^^^^^^^^^^^
146 |     pub use super::Executor;
147 |     pub use super::FromRow;
    |             ^^^^^^^^^^^^^^
...
151 |     pub use super::Type;
    |             ^^^^^^^^^^^

For more information about this error, try `rustc --explain E0432`.
error: could not compile `sqlx` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...

```

**Comment by JunioCalu** [comment]
> sqlx

@jb-alvarado @fooryo I use Manjaro, which is a distribution of Arch Linux. In practice, they're almost the same; if a package is compiled in Manjaro, it can also be compiled in Arch Linux. This compilation error is recent and specific to Arch Linux and Manjaro, and it's related to the "sqlx" crate. I think this problem can only be solved with a new update of the sqlx crate, or a patch specifically for it. Regarding the Rust update issue, I configured the AUR's PKGBUILD script to download a new stable TOOLCHAIN with each new compilation, changing the environment variables (RUSTUP_HOME and CARGO_HOME) to install this TOOLCHAIN in the package's compilation directory with each new build, so it's always up to date.

**Comment by fooryo** [comment]
@JunioCalu is there something I can directly do here for the sqlx thing? Or I have to wait for something? I have already updated my rust toolchain and still the sqlx errors happen.

**Comment by JunioCalu** [comment]
> @JunioCalu is there something I can directly do here for the sqlx thing? Or I have to wait for something? I have already updated my rust toolchain and still the sqlx errors happen.

I'll be conducting some tests on Manjaro 23.1.4, and then I'll circle back with a response to this.

**Comment by jb-alvarado** [comment]
Somehow others has same problem on arch based systems: https://github.com/launchbadge/sqlx/issues/3149

**Comment by JunioCalu** [comment]
> @JunioCalu is there something I can directly do here for the sqlx thing? Or I have to wait for something? I have already updated my rust toolchain and still the sqlx errors happen.

@fooryo The issue has been resolved. I just found out that the sqlx crate does not support the build flag "-flto=auto" (Link Time Optimization, LTO) which is enabled by default by the Arch Linux package build system (makepkg tool). I've updated the AUR PKGBUILD to disable this compile flag during the build of ffplayout, by setting the build option "options=('!lto')". Consequently, the package now builds normally, without errors.

I've also updated the ffmpeg-ffplayout package, so it should now be building without any issues as well.

Remove any build cache related to your AUR helper (paru or yay) and initiate a new build with "paru -S ffplayout ffmpeg-ffplayout --noconfirm", or something similar.
